### PR TITLE
Symmetry corrections

### PR DESCRIPTION
**Pull request by goldmanm**
This PR expands RMG's ability to estimate symmetry numbers for resonance structures and cyclic compounds.

1. Symmetry numbers used in thermo come from Species objects, not molecule objects
1. When estimating the symmetry number of a Species, a hybrid resonance structure is created that is the average of all the bond orders of various resonance structures.
1. Redid how RMG calculates cyclic symmetry numbers, fixing #858, #119 (and reducing computational time by 50%).
1. Allow for float inputs to Adjacency Lists for the bond order (in addition to the strings). 
1. Updated to a newer cython version for increased functionality
1. Allowed for partial bonds to be represented with dots in ipython notebooks
1. Some doc-string fixes and other functionality improvements.

### CODE REVIEW

**Comment by goldmanm** [comment]
I updated the [ipython notebook that tracks cyclic symmetry errors](https://github.com/goldmanm/rmg_walkthroughs/blob/master/issues/symmetry%20cyclic%20symmetry.ipynb) to the changes in this branch, so you can easily see the successes and failures of the updated algorithm (many of which currently fail on our master branch now). The cyclic symmetry algorithm seems to assumes that the atoms are on a 2D plane. You can check out it's performance on a large variety of mono and bicyclics. @KEHANG, I also included the two bicyclics you were interested in (at the bottom of the section 'let's look at many examples'

**Comment by mliu49** [comment]
I think this will conflict with #980, which reorganized thermoTest.py.

**Comment by goldmanm** [comment]
@mjliu I can reorder the conflicts after #980 is merged.

**Comment by goldmanm** [comment]
NOTE: I will add source for deriving symmetry numbers for the test methods.

File: rmgpy/speciesTest.py [hunk 1]
```diff
@@ -149,6 +149,31 @@ def testResonaceIsomersRepresented(self):
...
+        Tests that getResonanceHybrid returns an isomorphic structure
+        which has intermediate bond orders
+        """
+        spec = Species().fromSMILES('C=C[CH]CC')
+        hybridMol = spec.getResonanceHybrid()
+        
+        self.assertTrue(hybridMol.toSingleBonds().isIsomorphic(spec.molecule[0].toSingleBonds()))
+        
+        # a rough check for intermediate bond orders
+        expected_orders = [1,1.5]
```
Comment (skrsna) [inline rmgpy/speciesTest.py]: Please add a comment about the source of this `expected_orders`

File: rmgpy/molecule/symmetryTest.py [hunk 2]
```diff
@@ -272,6 +272,14 @@ def testTotalSymmetryNumberBenzene(self):
         symmetryNumber = species.getSymmetryNumber()
         self.assertEqual(symmetryNumber, 12)
 
+    def testTotalSymmetryNumberbute_di_yl(self):
+        """
+        Test the Species.getSymmetryNumber() (total symmetry) on [CH2][CH]C=C
+        """
+        species = Species().fromSMILES('[CH2][CH]C=C')
+        symmetryNumber = species.getSymmetryNumber()
+        self.assertEqual(symmetryNumber, 2)
```
Comment (skrsna) [inline rmgpy/molecule/symmetryTest.py]: Please add the source for the original symmetry number as a comment.

File: rmgpy/molecule/vf2Test.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        self.mol = Molecule().fromSMILES("CC(=O)C[CH2]")
+        self.mol2 = self.mol.copy(deep=True)
+
+    def test_import_graph(self):
+        """Test that we can add graphs to the object and that they are sorted"""
+
+        self.mol.sortVertices()
+        
+        ordered_original_connectivity_order = [getVertexConnectivityValue(atom) for atom in self.mol.atoms]
+        unordered_original_connectivity_order = [getVertexConnectivityValue(atom) for atom in self.mol.atoms]
```
Comment (skrsna) [inline rmgpy/molecule/vf2Test.py]: Please delete this piece of code.

File: rmgpy/speciesTest.py (hunk 1 above)
Comment (goldmanm) [inline rmgpy/speciesTest.py]: added! thanks

File: environment_linux.yml [hunk 4]
```diff
@@ -8,7 +8,7 @@ dependencies:
   - numpy >=1.10.0
   - cantera >=2.2
   - coolprop
-  - cython ==0.21
+  - cython ==0.25.2
```
Comment (skrsna) [inline environment_linux.yml]: Please use `>=` instead of `==`

File: rmgpy/molecule/graphTest.py [hunk 5]
```diff
@@ -640,6 +640,64 @@ def test_getPolycyclicRings(self):
         for ring in expectedContinuousRings:
             self.assertTrue(ring in continuousRings)
 
+
+    def test_getLargestRing(self):
+        """
+        Test that the Graph.getPolycyclicRings() method returns only polycyclic rings.
+        """
+        vertices = [Vertex() for i in range(27)]
+        bonds = [
```
Comment (skrsna) [inline rmgpy/molecule/graphTest.py]: Please add a comment of what this molecule name is or where it came from. Thanks

File: rmgpy/molecule/symmetry.py [hunk 6]
```diff
@@ -344,89 +344,78 @@ def calculateCyclicSymmetryNumber(molecule):
...
     
-    rings = molecule.getSmallestSetOfSmallestRings()
-
-    # Get symmetry number for each ring in structure
-    for ring0 in rings:
-
-        # Make another copy structure
-        structure = molecule.copy(True)
-        ring = [structure.atoms[molecule.atoms.index(atom)] for atom in ring0]
+    # for polycyclics, We should be getting the largest ring, not the smallest
```
Comment (skrsna) [inline rmgpy/molecule/symmetry.py]: Please add the reason(why?) for the above comment.

**Comment by skrsna** [comment]
I think you guys can merge this pull request.

**Pull request merged**